.github/workflows/build.yml

```diff
@@ -2,6 +2,7 @@ name: Build and Check
 
 on:
   push:
+  merge_group:
```

Comment:
I have found out that when using merge queues, you should run workflows on `pull_request` or `merge_group` for any branch but add `push` only for your default branches (main in our case). This works because whenever you push to a branch it will synch and run the tests and you do not get duplicated runs because of the push event that is not restricted to main. Let me know if it is not clear.

Reply:
Something like this would work then:
```
on:
  push:
    branches:
      - main
  merge_group:
```